### PR TITLE
setup circle ci / fix coveralls

### LINKED ISSUE #344: coverage reporting from travis is broken
e.g.:  https://travis-ci.org/sul-dlss/dor_indexing_app/builds/645671844#L1291-L1306

```
Coverage report generated for RSpec to /home/travis/build/sul-dlss/dor_indexing_app/coverage. 340 / 353 LOC (96.32%) covered.
Coveralls encountered an exception:
NoMethodError
undefined method `coverage' for #<SimpleCov::SourceFile:0x0000000005deda28>
Did you mean?  coverage_data
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/coveralls-0.7.1/lib/coveralls/simplecov.rb:50:in `block in format'
/home/travis/.rvm/rubies/ruby-2.5.3/lib/ruby/2.5.0/forwardable.rb:229:in `each'
/home/travis/.rvm/rubies/ruby-2.5.3/lib/ruby/2.5.0/forwardable.rb:229:in `each'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/coveralls-0.7.1/lib/coveralls/simplecov.rb:40:in `format'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov/formatter/multi_formatter.rb:10:in `block in format'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov/formatter/multi_formatter.rb:8:in `map'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov/formatter/multi_formatter.rb:8:in `format'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov/result.rb:49:in `format!'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov/configuration.rb:196:in `block in at_exit'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov.rb:201:in `run_exit_tasks!'
/home/travis/build/sul-dlss/dor_indexing_app/vendor/bundle/ruby/2.5.0/gems/simplecov-0.18.1/lib/simplecov/defaults.rb:31:in `block in <top (required)>'
```

The build (at least for #343, where i noticed this) correctly reports as passed on the github UI, but GH (unsurprisingly) no longer shows coverage stats.

### LINKED ISSUE #542: move CI build from travis to circleci
See also #344 about coverage -- make sure coverage works.

- get build working in circleCI
- with code coverage
- update badges in README (build, coverage)
- stop doing travis builds

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Fixes #344 and fixes #542 - switch from travis to circle ci, fix coveralls reporting

NOTE: after merge, updating the branch protection rules to disable travis and enable circle ci build: https://github.com/sul-dlss/dor_indexing_app/settings/branch_protection_rules/414234

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-cache: bundler
-rvm:
-  - 2.7.1
-
-before_install:
-  - gem update --system
-  - npm install -g openapi-enforcer-cli
-
-script:
-  - ./validate-openapi.bash || travis_terminate 1
```
Comment (mjgiarlo) [inline .travis.yml:11]: Do you think this PR could also remove `validate-openapi.bash`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            }
+      - run:
+          name: Check styles using rubocop
+          command: bundle exec rubocop
+      - run:
+          name: Setup Code Climate test-reporter
+          command: |
+            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+            chmod +x ./cc-test-reporter
+            ./cc-test-reporter before-build
```
Comment (mjgiarlo) [inline .circleci/config.yml:45]: Have you already set `CC_TEST_REPORTER_ID` in the CircleCI env for DIA?

File: .travis.yml (hunk 1 above)
Comment (peetucket) [reply .travis.yml:11]: yes, good catch - just deleted the file

File: .circleci/config.yml (hunk 2 above)
Comment (peetucket) [reply .circleci/config.yml:45]: Yes, I did that

Comment (mjgiarlo) [reply .circleci/config.yml:45]: thank you!

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**